### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by Stezido**
This PR updates following dependencies:
`fastify`: from `1.14.4` to `3.3.0`
`tar-fs`: from `1.16.3` to `2.1.0`
`exceljs`: from `1.10.0` to `4.0.3`

Fastify has some breaking changes.
The other deps` breaking changes don't affect us.

The viewHistory endpoints' filter validations for publisher and eventType are added.

### CODE REVIEW

File: api/src/notification_list.ts [hunk 1]
```diff
@@ -287,84 +288,101 @@ async function getMetadata(
...
-      reply.status(400).send({
-        apiVersion: "1.0",
-        error: {
-          code: 400,
-          message: "if present, the query parameter `limit` must be a positive integer",
-        },
-      });
-      return;
-    }
+      // Default: last create history event
```
Comment (daniel-arnauer) [inline api/src/notification_list.ts:310]: // Default: last create**d** history event

File: api/src/project_view_history.ts [hunk 2]
```diff
@@ -144,30 +156,36 @@ export function addHttpHandler(server: FastifyInstance, urlPrefix: string, servi
...
-      if (isNaN(offset)) {
-        reply.status(400).send({
-          apiVersion: "1.0",
-          error: {
-            code: 400,
-            message: "if present, the query parameter `offset` must be an integer",
-          },
-        });
-        return;
+      // Default: last create history event
```
Comment (daniel-arnauer) [inline api/src/project_view_history.ts:159]: // Default: last **created** history event

File: api/src/project_view_history_v2.ts [hunk 3]
```diff
@@ -167,35 +179,56 @@ export function addHttpHandler(server: FastifyInstance, urlPrefix: string, servi
...
-      if (isNaN(offset)) {
-        reply.status(400).send({
-          apiVersion: "1.0",
-          error: {
-            code: 400,
-            message: "if present, the query parameter `offset` must be an integer",
-          },
-        });
-        return;
+      // Default: last create history event
```
Comment (daniel-arnauer) [inline api/src/project_view_history_v2.ts:182]: // Default: last **created** history event

File: api/src/service/domain/workflow/project_history_get.ts [hunk 4]
```diff
@@ -1,19 +1,18 @@
...
 
 export interface Filter {
-  publisher: Identity;
-  startAt: string; // ISO timestamp
-  endAt: string; // ISO timestamp
-  eventType: Identity;
+  publisher?: Identity;
+  startAt?: string; // ISO timestamp;
+  endAt?: string; // ISO timestamp;
+  eventType?: string;
```
Comment (daniel-arnauer) [inline api/src/service/domain/workflow/project_history_get.ts:15]: why has been the "?" operator added?

File: api/src/service/domain/workflow/subproject_history_get.ts [hunk 5]
```diff
@@ -1,19 +1,18 @@
...
 
 export interface Filter {
-  publisher: Identity;
-  startAt: string; // ISO timestamp
-  endAt: string; // ISO timestamp
-  eventType: Identity;
+  publisher?: Identity;
+  startAt?: string; // ISO timestamp
+  endAt?: string; // ISO timestamp
+  eventType?: string;
```
Comment (daniel-arnauer) [inline api/src/service/domain/workflow/subproject_history_get.ts:15]: why has been the "?" operator added?

File: api/src/service/domain/workflow/workflowitem_history_get.ts [hunk 6]
```diff
@@ -1,19 +1,18 @@
...
 
 export interface Filter {
-  publisher: Identity;
-  startAt: string; // ISO timestamp
-  endAt: string; // ISO timestamp
-  eventType: Identity;
+  publisher?: Identity;
+  startAt?: string; // ISO timestamp
+  endAt?: string; // ISO timestamp
+  eventType?: string;
```
Comment (daniel-arnauer) [inline api/src/service/domain/workflow/workflowitem_history_get.ts:15]: why has been the "?" operator added?

File: api/src/service/domain/workflow/subproject_history_get.ts [hunk 7]
```diff
@@ -50,15 +49,17 @@ export const getHistory = async (
...
+  if (filter.startAt !== undefined) {
+    const startAt = filter.startAt;
     subprojectTraceEvents = subprojectTraceEvents.filter(
-      (event) => new Date(event.businessEvent.time) >= new Date(filter.startAt),
+      (event) => new Date(event.businessEvent.time) >= new Date(startAt),
     );
   }
 
   if (filter.endAt) {
+    const endAt = filter.endAt;
```
Comment (daniel-arnauer) [inline api/src/service/domain/workflow/subproject_history_get.ts:60]: `if (filter.startAt)` is fine

File: api/src/service/domain/workflow/project_history_get.ts (hunk 4 above)
Comment (Stezido) [reply api/src/service/domain/workflow/project_history_get.ts]: I've done the "filter my not be a empty object" for project.viewhistory.

File: frontend/src/api.js [hunk 8]
```diff
@@ -1,5 +1,6 @@
 import axios from "axios";
 import _isEmpty from "lodash/isEmpty";
+import isEmpty from "lodash/isEmpty";
```
Comment (daniel-arnauer) [inline frontend/src/api.js:4]: `_isEmpty ` and `isEmpty ` are the same

File: e2e-test/package.json [hunk 9]
```diff
@@ -19,7 +19,7 @@
   "main": "index.js",
   "devDependencies": {
     "accounting": "^0.4.1",
-    "cypress": "^4.11.0",
+    "cypress": "^4.12.1",
```
Comment (daniel-arnauer) [inline e2e-test/package.json:22]: updating to newest version (5.1.0) is also possible

**Review by daniel-arnauer — APPROVED** [review]
approved

**Pull request merged**